src/tests/ftest/launch.py

```diff
@@ -239,11 +285,77 @@ def get_test_files(test_list):
     test_files = [{"py": test, "yaml": None} for test in test_list]
     for test_file in test_files:
         base, _ = os.path.splitext(test_file["py"])
-        test_file["yaml"] = "{}.yaml".format(base)
+        test_file["yaml"] = replace_yaml_file(
+            "{}.yaml".format(base), args, tmp_dir)
 
     return test_files
 
 
+def replace_yaml_file(yaml_file, args, tmp_dir):
+    """Replace the server/client yaml file placeholders.
+
+    Replace any server or client yaml file placeholder names with the host
+    names provided by the command line arguments in a copy of the original
+    test yaml file.  If no replacements are specified return the original
+    test yaml file.
+
+    Args:
+        yaml_file (str): test yaml file
+        args (argparse.Namespace): command line arguments for this program
+        tmp_dir (TemporaryDirectory): temporary directory object to use to
+            write modified yaml files
+
+    Returns:
+        str: the test yaml file
+
+    """
+    if args.test_servers:
+        # Determine which placeholder names need to be replaced in this yaml by
+        # getting the lists of hosts specified in the yaml file
+        unique_hosts = {"servers": set(), "clients": set()}
+        for key, placeholders in find_yaml_hosts(yaml_file).items():
+            if key in SERVER_KEYS:
+                unique_hosts["servers"].update(placeholders)
+            elif key in CLIENT_KEYS:
+                # If no specific clients are specified use a specified server
+                key = "clients" if args.test_clients else "servers"
+                unique_hosts[key].update(placeholders)
+
+        # Map the placeholder names to values provided by the user
+        mapping_pairings = [("servers", args.test_servers.split(","))]
```

Comment:
FWIW, I'd really love to handle node (list) arguments to our tools with _cluset -e_ so that we could accept any valid clustershell nodeset specification.  I actually tried to do this with the nodelist passed to `ftest.sh`.  I quickly discovered cea-hpc/clustershell#410 however.  :frowning_face:

Reply:
That's unfortunate.  We lose the order with NodeSet as well:
```
>>> a = NodeSet.fromlist(["wolf-100vm3","wolf-122","wolf-123","wolf-124","wolf-125"])
>>> str(a)
'wolf-[122-125],wolf-100vm3'
>>> list(a)
['wolf-122', 'wolf-123', 'wolf-124', 'wolf-125', 'wolf-100vm3']

```

Reply:
> That's unfortunate. We lose the order with NodeSet as well:

Which is surely what is underlying `cluset` so not surprising.  But a pity indeed.

> But I guess that is the nature of a `set`

I guess.  Still unfortunate.  At least the ticket is still open.  You could go +1 it.  :smile: